### PR TITLE
fix: Force PLATFORMS environment variable when we build Elastic Agent dependencies on arm64

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

It forces the `PALFORMS` env var when we build the Elastic Agent beats dependencies in an arm64 host.

## Why is it important?

The current code tries to build amd64 packages in ARM architecture, we only support cross-compile for all architectures on amd64 machines. This causes that the mage package fails when you make it on an ARM host. The change in this PR build only amd64 binary dependencies in arm64.

## Checklist

- [X] My code follows the style guidelines of this project
- [ X I have commented my code, particularly in hard-to-understand areas
~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~
~- [ ] I have added tests that prove my fix is effective or that my feature works~
~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~

## Author's Checklist

- [ ]

## How to test this PR locally

* Launch a ubuntu-18.04/amd64 VM 
* Enter on x-pack/elastic-agent
* Run mage package
* It will build filebeat, metricbeat, heartbeat, and elastic-agent

* Launch a ubuntu-18.04/arm64 VM 
* Enter on x-pack/elastic-agent
* Run mage package
* It will build Elastic-Agent only

## Related issues

- closes https://github.com/elastic/beats/issues/26246

### CODE REVIEW

**Comment by kuisathaverat** [comment]
Greeeeeeen!!!!!

Screenshot 2021-06-23 at 12 14 01

**Review by mdelapenya — APPROVED** [review]

File: x-pack/elastic-agent/magefile.go [hunk 1]
```diff
@@ -579,6 +579,10 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 		}
 
 		os.Setenv(agentDropPath, dropPath)
+		if runtime.GOARCH == "arm64" {
+			os.Setenv("PLATFORMS", runtime.GOOS+"/"+runtime.GOARCH)
+			defer os.Unsetenv("PLATFORMS")
```
Comment (v1v) [inline x-pack/elastic-agent/magefile.go:584]:
```suggestion
			defer os.Unsetenv("PLATFORMS")
			os.Setenv("PLATFORMS", runtime.GOOS+"/"+runtime.GOARCH)
```

**Review by cachedout — APPROVED** [review]

File: x-pack/elastic-agent/magefile.go (hunk 1 above)
Comment (kuisathaverat) [reply x-pack/elastic-agent/magefile.go:584]: it is a deferred unset, it would happen at the end of the function.

**Review by v1v — APPROVED** [review]
+1

File: x-pack/elastic-agent/magefile.go [hunk 2]
```diff
@@ -579,6 +579,10 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
...
+		}
```
Comment (urso) [inline x-pack/elastic-agent/magefile.go:585]: Maybe we don't want to unset the variable if it was set.

e.g.

```
const platformsVar = "PLATFORMS"
oldPlatforms := os.GetEnv(platformsVar)
os.Setenv(...)
if oldPlatforms != "" {
  defer os.Setenv(platformsVar, oldPlatform)
} else {
  defer os.Unsetenv(platofrmsVar)
}
```

Comment (kuisathaverat) [reply x-pack/elastic-agent/magefile.go:585]: in arm64 is not defined ever, in any case, I will add the backup logic

**Review by urso — APPROVED** [review]

**Pull request merged**